### ISSUE TITLE
Reports not reporting out of stock products

### ISSUE THREAD

**Issue by apb1963** [woocommerce/woocommerce#16398]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ x] The issue still exists against the latest `master` branch of WooCommerce on Github
- [ x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Import new item with zero quantity stock using built-in CSV importer.
2. Go to Reports>Stock>All 3 tabs
3.

## Expected/actual behavior

When I follow those steps, I see...
Nothing.  The product is not listed on any of the 3 tabs.  Low, Out of Stock or Most Stocked.
I was expecting to see...
A product listed as out of stock.

## Isolating the problem

- [ x] This bug happens with only WooCommerce plugin active
- [ x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [ x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
`
### WordPress Environment ###

Home URL: https://greetonix.com
Site URL: https://greetonix.com
WC Version: 3.1.1
Log Directory Writable: ✔
WP Version: 4.8.1
WP Multisite: –
WP Memory Limit: 96 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.21
PHP Post Max Size: 6 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.38.0
OpenSSL/1.0.1t

SUHOSIN Installed: –
MySQL Version: ❌ 5.5.57 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.1.1
WC Database Prefix: [redacted]
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (9) ###

affilinet Performance Ads: by affilinet – 1.8.7
Akismet Privacy Policies: by Inpsyde GmbH – 1.1.2
Akismet Anti-Spam: by Automattic – 3.3.4
Code Snippets: by Shea Bunge – 2.8.6
Contact Form 7: by Takayuki Miyoshi – 4.8.1
TinyMCE Advanced: by Andrew Ozz – 4.6.3
Woocommerce Table Rate Shipping: by JEM Plugins – 1.2.3
WooCommerce PayPal Powered by Braintree Gateway: by WooCommerce – 2.0.4
WooCommerce: by Automattic – 3.1.1

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #5 - /
Cart: #6 - /cart
Checkout: #7 - /checkout
My account: #8 - /my-account

### Theme ###

Name: Storefront
Version: 2.2.4
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
@apb1963 `stock` is not the only property that would need to be set. `manage_stock` would also need to be true for the item, otherwise the stock level is not considered.

**Comment by apb1963** [comment]
Stock management is enabled in Woo>Products>Inventory. Are you saying that's not sufficient?  If not, then what does that setting do?   Thank you

**Comment by mikejolley** [comment]
Got it. See https://github.com/woocommerce/woocommerce/pull/16408

**Comment by apb1963** [comment]
Sorry but I'm new to this.  I followed the link, it noted 2 test failures.  Do I just wait until you guys take care of it and it will all automagically propagate as an update when you're done, or is there something I'm supposed to do?  

Some additional things I've noticed.  Neither stock nor stock_status have sample values; presumably it's because my first product listed has zero stock but I haven't verified the reason.  More importantly, I added manage_stock to my CSV file, but it's not selectable from the menu; i.e. it doesn't exist on the menu, so I can't import it.

Thank you
